### PR TITLE
fix exceptions with scrolling in the break point condition completion window

### PR DESCRIPTION
**Pull request by JieCarolHu**
### Bugs this fixes
https://github.com/dotnet/roslyn/issues/23890
https://github.com/dotnet/roslyn/issues/23891

### CODE REVIEW

**Comment by JieCarolHu** [comment]
@dpoeschl  Could you please review this PR? Thanks

File: src/EditorFeatures/Core.Wpf/Completion/Presentation/ToolTipProvider.cs [hunk 1]
```diff
@@ -67,14 +67,17 @@ public CancellableContentControl(ToolTipProvider toolTipProvider, CustomCommitCo
...
                 this.Content = toolTipProvider._defaultTextBlock;
 
-                // Kick off the task to produce the new content.  When it completes, call back on 
+                // Kick off the task to produce the new content.  When it completes, call back on
                 // the UI thread to update the display.
-                var scheduler = TaskScheduler.FromCurrentSynchronizationContext();
-                item.GetDescriptionAsync(_cancellationTokenSource.Token)
-                              .ContinueWith(ProcessDescription, _cancellationTokenSource.Token,
+                var getDescTask = item.GetDescriptionAsync(_cancellationTokenSource.Token);
+                if (getDescTask != null)
```
[the remark this answers is not in the document]
Comment (JieCarolHu) [reply src/EditorFeatures/Core.Wpf/Completion/Presentation/ToolTipProvider.cs]: udpated

File: src/EditorFeatures/Core/Implementation/IntelliSense/ViewTextSpan.cs [hunk 2]
```diff
@@ -133,6 +133,8 @@ private static Span MapDownToSnapshot(Span span, IProjectionSnapshot start, ITex
                     }
                 }
             }
+
+            return default;
```
[the remark this answers is not in the document]
Comment (JieCarolHu) [reply src/EditorFeatures/Core/Implementation/IntelliSense/ViewTextSpan.cs]: added Contract.Failed() as we should never reach here

File: src/EditorFeatures/Core/Implementation/IntelliSense/ViewTextSpan.cs [hunk 3]
```diff
@@ -118,7 +118,7 @@ private static IEnumerable<Span> MapUpToSnapshotRecursive(SnapshotSpan start, IP
         private static Span MapDownToSnapshot(Span span, IProjectionSnapshot start, ITextSnapshot target)
         {
             var sourceSpans = new Queue<SnapshotSpan>(start.MapToSourceSnapshots(span));
-            while (true)
+            while (sourceSpans.Count > 0)
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Implementation/IntelliSense/ViewTextSpan.cs]: I'm now confused by this change -- doesn't this just mean we replace one exception (the .Dequeue on empty queue) with the Contract.Fail now?

File: src/EditorFeatures/Core.Wpf/Completion/Presentation/RoslynCompletionSet.cs [hunk 4]
```diff
@@ -172,13 +172,16 @@ private VSCompletion GetVSCompletion(CompletionItem item, string displayText = n
...
-            // Linear search is ok since this is only called by the user manually selecting 
+            // Linear search is ok since this is only called by the user manually selecting
             // an item.  Creating a reverse mapping uses too much memory and affects GCs.
             foreach (var kvp in CompletionItemMap)
             {
                 if (kvp.Value == completion)
                 {
-                    return kvp.Key;
+                    // return null if the CompletionItem.Document is null
+                    // which means it reachs the text user types on the top of the completion list in the breakpoint condition window
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core.Wpf/Completion/Presentation/RoslynCompletionSet.cs]: I would mind if this comment better explained what is ultimately breaking if we don't return null here. Put another why: why is the null check here vs. whoever is calling this?

**Comment by JieCarolHu** [comment]
retest retest windows_release_unit32_prtest

**Comment by JieCarolHu** [comment]
retest windows_debug_vs-integration_prtest

**Comment by JieCarolHu** [comment]
retest windows_release_unit32_prtest

**Comment by JieCarolHu** [comment]
retest windows_release_vs-integration_prtest

**Comment by tmat** [comment]
Seems like it's gonna fix https://devdiv.visualstudio.com/DevDiv/NET%20Developer%20Experience%20Productivity/_workitems/edit/470724

**Comment by tmat** [comment]
Let's move to 15.8

File: src/EditorFeatures/Core.Wpf/Completion/Presentation/CustomCommitCompletion.cs [hunk 5]
```diff
@@ -57,7 +57,9 @@ public void Commit()
         public Task<CompletionDescription> GetDescriptionAsync(CancellationToken cancellationToken)
         {
             var service = CompletionService.GetService(this.CompletionItem.Document);
-            return service?.GetDescriptionAsync(this.CompletionItem.Document, this.CompletionItem, cancellationToken);
+            return service==null?
```
Comment (tmat) [inline src/EditorFeatures/Core.Wpf/Completion/Presentation/CustomCommitCompletion.cs]: Nit: missing spaces between `service`, `==`, '?`

**Comment by JieCarolHu** [comment]
@jinujoseph Could you approve this merge? Thanks

**Comment by jinujoseph** [comment]
Approved to merge for 15.8.Preview5 ( once merge pls reach out to DC customer)

**Comment by JieCarolHu** [comment]
retest windows_debug_vs-integration_prtest

**Comment by JieCarolHu** [comment]
retest windows_release_vs-integration_prtest

**Comment by JieCarolHu** [comment]
retest windows_debug_vs-integration_prtest

**Comment by JieCarolHu** [comment]
retest windows_debug_vs-integration_prtest

**Comment by JieCarolHu** [comment]
retest windows_release_vs-integration_prtest

**Comment by JieCarolHu** [comment]
retest windows_debug_vs-integration_prtest

**Comment by JieCarolHu** [comment]
retest windows_debug_vs-integration_prtest

**Comment by jinujoseph** [comment]
Taking one of the fixes via https://github.com/dotnet/roslyn/pull/28471

**Comment by jinujoseph** [comment]
@JieCarolHu closing this , pls open a new one with #23891 change alone

**Pull request closed without merging**